cmd/cmd.go

```diff
@@ -66,13 +58,67 @@ func init() {
 	rootCmd.Flags().BoolVar(&outputStdOut, "stdout", false,
 		"output schema content to stdout instead of a file and disable error output",
 	)
-	// TODO: implement
+
 	rootCmd.Flags().BoolVar(&disallowAdditionalProperties, "disallow-additional-properties", false,
 		"set additionalProperties to false in the generated schema and in nested objects",
 	)
+
+	rootCmd.Flags().BoolVar(&allowEmpty, "allow-empty", false,
+		"allow empty schema if no variables are found, otherwise error",
+	)
+
+	rootCmd.Flags().BoolVar(&requireAll, "require-all", false,
+		"require all variables to be present in the schema, even if a default value is specified",
+	)
+
+	rootCmd.Flags().StringVarP(&input, "input", "i", ".",
+		"input folder containing .tf files",
+	)
+
 	// TODO: implement
-	rootCmd.Flags().BoolVar(&allowEmpty, "allow-empty", false, "allow empty schema if no variables are found, otherwise error")
-	rootCmd.Flags().StringVarP(&input, "input", "i", ".", "input folder containing .tf files")
-	// TODO: implement
-	rootCmd.Flags().StringVarP(&output, "output", "o", "schema.json", "output file path for schema")
+	rootCmd.Flags().StringVarP(&output, "output", "o", "schema.json",
+		"output file path for schema",
+	)
+}
+
+func runCommand(cmd *cobra.Command, args []string) {
+	path, err := filepath.Abs(input) // absolute path
+	if err != nil {
+		errReturned = fmt.Errorf("could not get absolute path for %q: %w", input, err)
+
+		return
+	}
+
+	folder, err := os.Stat(path)
+	if err != nil {
+		errReturned = fmt.Errorf("could not open %q: %w", path, err)
+
+		return
+	}
+
+	if !folder.IsDir() {
+		errReturned = fmt.Errorf("input %q is not a directory", path)
+
+		return
+	}
+
+	output, err := jsonschema.CreateSchema(path, jsonschema.CreateSchemaOptions{
+		RequireAll:                requireAll,
+		AllowAdditionalProperties: !disallowAdditionalProperties,
+		AllowEmpty:                allowEmpty,
+	})
+	if err != nil {
+		errReturned = fmt.Errorf("error creating schema: %w", err)
+
+		return
+	}
+
+	jsonOutput, err := json.MarshalIndent(output, "", "    ")
+	if err != nil {
+		errReturned = fmt.Errorf("error marshalling schema: %w", err)
+
+		return
+	}
+
+	fmt.Println(string(jsonOutput))
```

Comment:
This is going to stdout (good default behaviour), so "-o" flag not implemented? (Or did I miss something?)